sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/SpecificParquetRecordReaderBase.java

```diff
@@ -144,13 +113,16 @@ public void initialize(InputSplit inputSplit, TaskAttemptContext taskAttemptCont
     String sparkRequestedSchemaString =
         configuration.get(ParquetReadSupport$.MODULE$.SPARK_ROW_REQUESTED_SCHEMA());
     this.sparkSchema = StructType$.MODULE$.fromString(sparkRequestedSchemaString);
-    this.reader = new ParquetFileReader(
-        configuration, footer.getFileMetaData(), file, blocks, requestedSchema.getColumns());
+    this.reader =  new ParquetFileReader(HadoopInputFile.fromPath(file, configuration),
+            HadoopReadOptions.builder(configuration).build());
```

Comment:
You can't use this because of the leak issue at https://github.com/apache/parquet-mr/pull/510. This was fixed in Parquet 1.11.

Reply:
This is the only constructor which is not deprecated available in parquet 1.10. Could you suggest any other alternative? Do we continue using the old way or is there another work around?

Reply:
I think the point is, don't change this until Parquet 1.11 is required. This should be part of that change. Are there any changes here that definitely work with the version of Parquet that Spark currently uses?

Reply:
ok.I shall change back to the old constructor. there are other api changes like for example ParquetFileReader.readFooter, ParquetFileReader.readAllFootersInParallel, deprecated class ParquetInputSplit